docs/authelia-from-scratch.md

```diff
@@ -0,0 +1,448 @@
+    proxy_set_header X-Forwarded-Proto $scheme;
+    proxy_set_header X-Forwarded-host authelia.example.com;
+    proxy_http_version 1.1;
+    proxy_set_header Upgrade $http_upgrade;
+    proxy_set_header Connection "upgrade";
+    proxy_cache_bypass $http_upgrade;
+    EOF
+
+    # cat <<EOF > ssl.conf
+    resolver 8.8.8.8 valid=300s;
+    resolver_timeout 10s;
+    ssl_certificate /etc/nginx/ssl/xxx.example.com.fullchain;
+    ssl_certificate_key /etc/nginx/ssl/xxx.example.com.key;
+    ssl_protocols TLSv1.2;
+    ssl_ciphers 'EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH';
+    ssl_ecdh_curve secp384r1;
+    ssl_prefer_server_ciphers on;
+    ssl_stapling on;
+    ssl_stapling_verify on;
+    ssl_session_timeout 24h;
+    ssl_session_cache shared:SSL:50m;
+    ssl_session_tickets off;
+    EOF
+
+### (re)Start NGINX
+    # ln sites-available/authelia.conf sites-enabled/
+    # systemctl restart nginx
+
+### Install Docker
+    # curl -fsSL https://download.docker.com/linux/debian/gpg | apt-key add -
+    # add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/debian $(lsb_release -cs) stable"
+    # apt-get update
+    # apt-get install docker-ce docker-ce-cli containerd.io
+
+### Install Docker Compose
+    # curl -s https://api.github.com/repos/docker/compose/releases/latest | grep browser_download_url | grep docker-compose-Linux-x86_64 | cut -d '"' -f 4 | wget -qi -
+    # chmod +x docker-compose-Linux-x86_64
+    # mv docker-compose-Linux-x86_64 /usr/local/bin/docker-compose
+
+### Install Node.JS
+    # cd /usr/src
+    # curl -sL https://deb.nodesource.com/setup_12.x | bash -
+    # apt-get install nodejs
+
+### Download, Build and Install Authelia
+    # cd /opt
+    # git clone https://github.com/clems4ever/authelia.git
+    # cd authelia
+    # source bootstrap.sh
+    # authelia-scripts build
+
+### Setup Authelia
+    # mkdir /etc/authelia/
+    # cat <<EOF > /etc/authelia/config.yml
+    ###############################################################
+    #                   Authelia configuration                    #
+    ###############################################################
+    
+    port: 9091
+    
+    logs_level: info
+    jwt_secret: a_secret
+    default_redirection_url: https://landing.example.com/
+    
+    totp:
+      issuer: landing.example.com
+    
+    authentication_backend:
+      ldap:
+        url: ldap://127.0.0.1
+        base_dn: dc=example,dc=com
+        additional_users_dn: ou=people
+        users_filter: uid={0}
+        additional_groups_dn: ou=groups
+        groups_filter: (&(member={dn})(objectclass=groupOfNames))
+        group_name_attribute: cn
+        mail_attribute: mail
+        user: cn=admin,dc=example,dc=com
+        password: $slapd-admin-password
+    
+    access_control:
+      default_policy: two_factor
+    
+    session:
+      name: authelia_session
+      secret: unsecure_session_secret
```

Comment:
same remark as above, why not use $session_secret as for other secrets?

Reply:
Good idea! Done :)